### ISSUE TITLE
Multiple -c annex.dotfile=true

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4271]
Spotted in the outputs of #4265 - no harm but ideally should be in a single instance. This is a build on current maint:

https://travis-ci.org/datalad/datalad/builds/660397445?utm_source=github_status&utm_medium=notification
```
[Level 3] Processing provided line 
[INFO   ] [2020-03-09 23:21:42.986949987] read: git ["--git-dir=./.git","--work-tree=.","--literal-pathspecs","-c","annex.dotfiles=true","-c","annex.dotfiles=true","-c","annex.dotfiles=true","-c","annex.dotfiles=true","-c","annex.dotfiles=true","-c","annex.dotfiles=true","show-ref","--hash","refs/heads/git-annex"] 
```

**Comment by kyleam** [comment]
Odd.  Given that we set this once in `AnnexRepo.__init__`, it seems there's likely a bug in how AnnexRepo handles that list.

https://github.com/datalad/datalad/blob/89030cb4ec6a4b08f580c1fd19548da75f485af6/datalad/support/annexrepo.py#L230

**Comment by kyleam** [comment]
I've had luck triggering it locally.

**Comment by kyleam** [comment]
The duplication is happening on git-annex's end.  I'll open a report later today.

<details>
<summary>script</summary>

```sh
#!/bin/sh

set -eu

cd "$(mktemp -d --tmpdir gx-XXXXXXX)"
git init >/dev/null

set -x
git annex version | head -1
git annex init -c foo.bar=true "$@" --debug 2>&1 | grep foo | tail -1
```

</details>



```diff
+ git annex version
+ head -1
git-annex version: 7.20190819
+ git annex init -c foo.bar=true --debug
+ grep foo
+ tail -1
[2020-03-10 12:05:40.798830979] call: git ["--git-dir=.git","--work-tree=.","--literal-pathspecs","-c","foo.bar=true","-c","foo.bar=true","-c","foo.bar=true","update-ref","refs/heads/git-annex","6fdfad9fb5dbe1e107340bc6c173c329a49f89aa"]
```

**Comment by kyleam** [comment]
[Filed][0] with git-annex.

[0]: https://git-annex.branchable.com/bugs/Some_calls_to_git_repeat_--config_values/

**Comment by kyleam** [comment]
For posterity: Joey fixed this in c8fec6ab0 (Fix a minor bug that caused options provided with -c to be passed multiple times to git, 2020-03-16).

**Comment by yarikoptic** [comment]
Thank you @kyleam!
```shell
$> ~/proj/datalad/datalad-master/tools/describegitannex c8fec6ab0
I: Head of the last commit:
  commit c8fec6ab031a25cca012f2f6858ef8251c484e50
  Author: Joey Hess <id@joeyh.name>
  Date:   Mon Mar 16 13:06:44 2020 -0400
  
      Fix a minor bug that caused options provided with -c to be passed multiple times to git.
  
  diff --git a/Annex.hs b/Annex.hs
  index 9e0bcd3d8..c914748d5 100644
  --- a/Annex.hs
  +++ b/Annex.hs

I: Markdown links to the commit
  [8.20200309-62-gc8fec6ab0](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=c8fec6ab0)
  > fixed in [8.20200309-62-gc8fec6ab0](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=c8fec6ab0) --[[yarikoptic]]
```
Thus - fixed in [8.20200309-62-gc8fec6ab0](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=c8fec6ab0)

**Comment by yarikoptic** [comment]
I still hope that by having such quick "excursions" into haskell I will get to read it better ;)
